staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -590,7 +590,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 			requestScope = "namespace"
 			namespaced = "Namespaced"
 		}
-		if strings.HasSuffix(action.Path, "/{path:*}") {
+		if strings.HasSuffix(action.Path, "/{name}") || strings.HasSuffix(action.Path, "/{path:*}") {
```

Comment:
More context is in: https://github.com/kubernetes/kubernetes/issues/95556

Re https://github.com/kubernetes/kubernetes/issues/95556#issuecomment-708431219 after thinking about this, I think that if a request is touching only a single object, it should be reported as scope "request" (not as scope "namespace").

Reply:
Re adding a test - I think that adding a test is potentially non-trivial here, and given that we have broken reporting at this point after #93523 exposed this bug, I would like to fix the metrics to ensure that we start tracking performance the right way and add a test independently.

Reply:
I probably agree with the classification.

However I can't agree with leaving this untested, this is kind of critical to the functioning of the system. Is there really not an existing test for this?! (I can't find one if there is...) It looks like apiserver_test.go has a framework which should make it not too hard to set up a test server and send various URL paths to it?

@deads2k

Reply:
> However I can't agree with leaving this untested, this is kind of critical to the functioning of the system.

Agree

> Is there really not an existing test for this?! (I can't find one if there is...)

The fact that it's currently broken indicates there is not ;-)

> It looks like apiserver_test.go has a framework which should make it not too hard to set up a test server and send various URL paths to it?

We have TestApiserverMetrics that looks like it does the setup already. It seems like we can make more requests to a variety of endpoints and verify the resulting metrics include all the dimensions/label values we'd expect.

Reply:
I didn't find any test too.

I don't see anything that is particularly useful in apiserver_test.

But I really like Jordan`s suggestion about the metrics test. Using this framework should actually be much less work.
I will try to get to it today.

[BTW - I'm glad that you agree with the fix at this point.]

Reply:
> I probably agree with the classification.

It seems reasonable to classify this way.  It's a little weird that it impacts the verb chosen though.